### PR TITLE
fix: Document view supports hotkeys

### PR DESCRIPTION
**Pull request by bk201-**
This pull request includes several changes to enhance the functionality and user experience of the document editor. The most important changes include adding support for keyboard shortcuts, refactoring the toolbar and dialog components, and updating dependencies.

### Enhancements and New Features:

* Added `react-hotkeys-hook` dependency to support keyboard shortcuts (`package.json`).
* Implemented keyboard shortcuts for saving, editing, and refreshing documents using `react-hotkeys-hook` in `DocumentPanel.tsx`. [[1]](diffhunk://#diff-0fcab1b22e15a6415b69b5fa02920458728ab54bedeafcfbbfbc8c52563cda43L9-R30) [[2]](diffhunk://#diff-0fcab1b22e15a6415b69b5fa02920458728ab54bedeafcfbbfbc8c52563cda43R100-R166)

### Refactoring and Code Improvements:

* Refactored `DocumentToolbar` to accept `onSave`, `onEdit`, and `onRefresh` as props and removed the inline dialog component, moving it to a separate file `UnsavedChangesDialog.tsx`. [[1]](diffhunk://#diff-8dd45d8cf96fd717d1fcee076f8ec50f141f100bb55dd775390766add5510114L6-R59) [[2]](diffhunk://#diff-8dd45d8cf96fd717d1fcee076f8ec50f141f100bb55dd775390766add5510114L126-R77) [[3]](diffhunk://#diff-794976f2336410a67b7393d3d8ef59ce9ff6088c32729ead395b4127c47abdf0R1-R46)

### Configuration Updates:

* Updated the target JavaScript version from `es2021` to `es2022` in `webpack.config.ext.js` and `webpack.config.views.js` to leverage new language features. [[1]](diffhunk://#diff-5d29680f6c2a7a02444ff9780dfd55a5627e9a21c2a9838485930218b70355e4L81-R81) [[2]](diffhunk://#diff-108f8faed781a9625577969e16f92713bc64a008c6833dd95512b9e31dde5b99L57-R57)

### CODE REVIEW

File: src/webviews/Document/DocumentToolbar.tsx [hunk 1]
```diff
@@ -3,115 +3,60 @@
...
-            setOpen(true);
-            setDoAction(() => async () => {
-                setOpen(false);
-                await dispatcher.refreshDocument();
-            });
-        } else {
-            void dispatcher.refreshDocument();
-        }
-    };
+    const onSaveHotkeyTitle = isMac ? 'Cmd+S' : 'Ctrl+S';
```
Comment (sevoku) [inline src/webviews/Document/DocumentToolbar.tsx:28]: Nit: on Mac the special keys are usually represented by their symbols:
* Command: ⌘ (U+2318)
* Shift: ⇧ (U+21E7)
image

File: package.json [hunk 2]
```diff
@@ -156,6 +156,7 @@
         "pg": "^8.13.0",
         "pg-connection-string": "^2.7.0",
         "pg-structure": "^7.15.3",
+        "react-hotkeys-hook": "^4.6.1",
```
Comment (tnaum-ms) [inline package.json:159]: a great discovery, thank you! I'll start using it for vCore as well!

File: src/webviews/Document/DocumentPanel.tsx [hunk 3]
```diff
@@ -6,21 +6,28 @@
...
     },
     container: {
         marginTop: '10px',
         height: 'calc(100% - 10px)', // 10px margin
+        display: 'flex',
+        flexDirection: 'column',
+        justifyContent: 'stretch',
+        alignItems: 'stretch',
+        gap: '5px',
     },
```
Comment (tnaum-ms) [inline src/webviews/Document/DocumentPanel.tsx:31]: did it help with monaco's height updates as well?

File: src/webviews/Document/DocumentToolbar.tsx (hunk 1 above)
Comment (bk201-) [reply src/webviews/Document/DocumentToolbar.tsx]: I've asked you recently how it works on Mac, you didn't response  :)

File: src/webviews/Document/DocumentPanel.tsx (hunk 3 above)
Comment (bk201-) [reply src/webviews/Document/DocumentPanel.tsx:31]: Probably yes. I've tested different ways to change size and it looks as work approach

File: package.json (hunk 2 above)
Comment (bk201-) [reply package.json:159]: There are some pitfalls there. For instance the library can stop default behavior but doesn't want to stop propagation, or doesn't catch key events from input elements by default

**Review by sevoku — APPROVED** [review]

**Pull request merged**